deepchecks/vision/checks/performance/image_segment_performance.py

```diff
@@ -148,14 +149,23 @@ def compute(self, context: Context, dataset_kind: DatasetKind) -> CheckResult:
                 # in order to return the bins object as the check result value
                 single_bin['metrics'] = _calculate_metrics(single_bin['metrics'], dataset)
                 single_bin['display_range'] = display_range
-                # For the plotly display need row per metric in the dataframe
-                for metric, val in single_bin['metrics'].items():
-                    display_data.append({'Metric': metric, 'Value': val, **bin_data})
+                # we don't show single columns in the display
+                if len(prop_bins) > 1:
+                    # For the plotly display need row per metric in the dataframe
+                    for metric, val in single_bin['metrics'].items():
+                        display_data.append({'Metric': metric, 'Value': val, **bin_data})
                 # Save for result
                 result_value[property_name].append(single_bin)
 
         display_df = pd.DataFrame(display_data)
-
+        if display_df.empty:
+            return CheckResult(value=dict(result_value))
+        first_metric = list(get_scorers_list(dataset, self.alternative_metrics).keys())[0]
```

Comment:
why not take metric column from the dataframe?

Reply:
not 100% sure if it would be first. probably will, doesn't matter too much, can change it